include/fmt/format.h

```diff
@@ -1552,13 +1553,18 @@ template <typename OutputIt, typename Char, typename UInt> struct int_writer {
   enum { sep_size = 1 };
 
   void on_num() {
+#if !defined(FMT_STATIC_THOUSANDS_SEPARATOR)
     std::string groups = grouping<Char>(locale);
+#else
+    constexpr const char kDefaultGroups[] = "\03";
+    basic_string_view<Char> groups(kDefaultGroups, sizeof(kDefaultGroups));
```

Comment:
You can probably remove https://github.com/fmtlib/fmt/blob/60c43e870387affb07de32ed4fc2794ac7a9f60d/include/fmt/format-inl.h#L209-L212 with this change.